db/kvstorewithcache.go

```diff
@@ -0,0 +1,125 @@
+package db
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/go-pkgs/cache"
+	"github.com/iotexproject/iotex-core/db/batch"
+)
+
+// kvStoreWithCache is an implementation of KVStore, wrapping kvstore with LRU caches of latest states
+type kvStoreWithCache struct {
+	store       KVStore
+	stateCaches map[string]*cache.ThreadSafeLruCache // map having lru cache to store current states for speed up
```

Comment:
i think the caches may need a lock to guard it

Reply:
ThreadSafeLruCache already has a lock in it "github.com/iotexproject/go-pkgs/cache", do you need another lock?

Reply:
added mutex for map!